core/lib/libtomcrypt/src/tee_ltc_provider.c

```diff
@@ -691,7 +691,7 @@ static void bn_clear(struct bignum *s)
 {
 	struct mpa_numbase_struct *bn = (struct mpa_numbase_struct *)s;
 
-	memset(bn, 0, bn->alloc);
+	memset(bn->d, 0, bn->alloc);
```

Comment:
Hello Etienne,

Thanks for your comment and research.

Indeed your proposal spots another issue of the `bignum` area.
But it is not the aim of the issue of this ticket.

Issue was the reset of the data which scratch also the metadatas. And the initial fix is only to reset datas instead of metadata, and that has no dependance with the length of the data (`bn->alloc`): length in bytes of the reset of the data is the same with or without your fix.
Your proposal fix the length of the data, without the reset of the data. The issue of this ticket is always present with your proposal and your fix.

So, i would prefer that, as you advise me few years ago, to split the issues and do a commit per subject. One which fix the scratch of the metadata, by the way, the initial proposal i made,
and another commit (order of commits is not a matter as subjects are not correlated), which fix the length (`bn->alloc`) with your proposal.

I have run the ST TEE nonreg with the initial fix, and the two fixes, and both are OK. (Tests are KO with your proposal only)

Best regards
Manu